sktime/libs/uni2ts/moirai_module.py

```diff
@@ -0,0 +1,135 @@
+from functools import partial
+
+from skbase.utils.dependencies import _check_soft_dependencies
+
+if _check_soft_dependencies("torch", severity="none"):
+    import torch
+    import torch.nn.functional as F
+    from torch import nn
+    from torch.distributions import Distribution
+    from torch.utils._pytree import tree_map
+
+if _check_soft_dependencies("huggingface_hub", severity="none"):
+    from huggingface_hub import PyTorchModelHubMixin
+
+if _check_soft_dependencies("hydra-core", severity="none"):
+    from hydra.utils import instantiate
+
+if _check_soft_dependencies(["jax", "jaxtyping"], severity="none"):
```

Comment:
Try to remove the usage of all types. To mitigate needing this as a soft-dependency.